pkg/storegateway/gateway_blocks_http.go

```diff
@@ -156,9 +207,45 @@ func (s *StoreGateway) BlocksHandler(w http.ResponseWriter, req *http.Request) {
 		ShowDeleted: showDeleted,
 		ShowSources: showSources,
 		ShowParents: showParents,
+		ActionType:  action,
 	}, blocksPageTemplate, req)
 }
 
+func (s *StoreGateway) performActionsOnBlocks(tenantID string, req *http.Request, action ActionType, blockUlids []string) error {
```

Comment:
I see how convenient is to implement it this way, but I think this is the first operation where store-gateway isn't a reader of block storage, but also a writer of it.

Are we good with that?

(I think this better fits compactor, who's the one writing and dealing with compaction, but I also understand the inconvenience of having it in a different component).

Reply:
as discussed in the weekly, we can leave this in store-gateway for now, if in the future we want to remove the write permission of store-gateway, I think this is acceptable since we are doing revertable changes, upload/delete no-compact marker is not a critical action.